controllers/cassandrabackup_controller.go

```diff
@@ -80,20 +80,41 @@ func (r *CassandraBackupReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 		}
 	}
 
+	// If there is anything in progress, simply requeue the request
+	if len(backup.Status.InProgress) > 0 {
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
 	if backup.GetDeletionTimestamp() != nil {
 		if ctrlutil.ContainsFinalizer(backup, finalizerName) {
-			// Backup is being deleted, call Medusa to remove it also from storage
-			err = r.removeBackup(ctx, backup)
-			if err != nil {
-				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
-			}
+			// Deletion process has already been started, return later
+			patch := client.MergeFrom(backup.DeepCopy())
+			backup.Status.InProgress = append(backup.Status.InProgress, "removeBackup")
 
-			// Delete the finalizer to allow deletion process to finish
-			ctrlutil.RemoveFinalizer(backup, finalizerName)
-			err = r.Update(ctx, backup)
-			if err != nil {
-				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
+			if err := r.Status().Patch(context.Background(), backup, patch); err != nil {
+				r.Log.Error(err, "failed to patch status with backup deletion start time", "Backup", backup)
+				return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, err
 			}
+
+			go func() {
+				// Backup is being deleted, call Medusa to remove it also from storage
+				err = r.removeBackup(ctx, backup)
+				if err != nil {
+					r.Log.Error(err, "failed to remove backup", "Backup", backup)
+					backup.Status.InProgress = removeValue(backup.Status.InProgress, "removeBackup")
+					if err := r.Status().Patch(context.Background(), backup, patch); err != nil {
+						r.Log.Error(err, "failed to patch status", "Backup", backup)
+					}
+				}
+
+				// Delete the finalizer to allow deletion process to finish
+				ctrlutil.RemoveFinalizer(backup, finalizerName)
```

Comment:
If `removeBackup` fails we still clear the finalizer here and the CassandraBackup is subsequently deleted. The user could be completely unaware that some or all of the back in remote storage still exists. I think we should either

* Continue retrying the operation and/or
* Update the status to indicate that the operation failed